File: CardMasterCard/Cards/Texture.vb
Imports System.Drawing
Imports System.IO

Namespace Card

    Public Class Texture

        Private m_name As String = Nothing
        Private m_color As String = Nothing

        Public Sub New()
            Reset()
        End Sub

        Public Sub New(name As String)
            Me.Name = name
        End Sub

        Public Sub New(name As String, color As String)
            Me.Name = name
            Me.Color = color
        End Sub

        Public Property Name As String
            Get
                Return m_name
            End Get
            Set(value As String)
                m_name = value
            End Set
        End Property

        Public Property Color As String
            Get
                Return m_color
            End Get
            Set(value As String)
                m_color = value
            End Set
        End Property

        Public Overrides Function ToString() As String
            Return Me.Name
        End Function

    End Class

End Namespace

File: CardMasterCard/Cards/Texture.cs
using System;

namespace CardMasterCard.Card
{
    public class Texture
    {
        public String Name { get; set; } = null;
        public String Color { get; set; } = null;

        public Texture()
        { }

        public Texture(String name)
        {
            this.Name = name;
        }

        public Texture(String name, String color)
        {
            this.Name = name;
            this.Color = color;
        }

        public override String ToString()
        {
            return this.Name;
        }

    }
}
